### PR TITLE
Convert old clock triggers to wall_clock xtriggers.

### LINKED ISSUE #5285: clock-triggers: re-implement as xtriggers
Cylc currently provides two mechanisms for clock-limiting tasks:

* [[xtriggers]wall_clock](https://cylc.github.io/cylc-doc/stable/html/reference/config/workflow.html#flow.cylc[scheduling][xtriggers]%3Cxtrigger%20name%3E)
  The new approach, the `@clock` trigger is used in the graph like regular dependencies. This has the advantage of transparency.
* [[special tasks]clock-trigger](https://cylc.github.io/cylc-doc/stable/html/reference/config/workflow.html#flow.cylc[scheduling][special%20tasks]clock-trigger)
  The old approach. marked as deprecated but still widely used. Allows listing of tasks which should be clock-limited.

We will presumably look to retire the old `clock-triggers` at some point after Cylc 7 back-compat mode is dropped, but this is still a way off. This means we need to keep these triggers working for the near future (which is something else to go wrong #5217).

Suggest converting clock-triggers to xtriggers at configure time. I think we can do this fairly easily at configure time something along the lines of:

```python
# config.py
def convert_clock_triggers(self):
    for string in self.cfg['scheduling']['special tasks']['clock-trigger']:
        task_name, offset  = do_something(string)
        xtrig_name = f'_cylc_clock_trigger_{task_name}'
        self.cfg['scheduling']['xtriggers'][xtrig_name] = f'wall_clock({offset})'
        taskdef = self.taskdefs[task_name]
        taskdef.add_xtrig_label(xtrig_name)
```

This way the old triggers will continue working under the new logic allowing us to retire this feature early. The old triggers will also become visible in the UI once xtrigger support is added.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by hjoliver**
Close #5285 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened https://github.com/cylc/cylc-doc/pull/575
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

Covered by existing tests:
```console
$ git grep 'clock-trigger' tests/*functional
tests/flakyfunctional/events/40-stall-despite-clock-trig/flow.cylc:        clock-trigger = t1(P0D)
tests/flakyfunctional/special/04-clock-triggered/flow.cylc:        clock-trigger = clock{% if OFFSET is defined %}({{OFFSET}}){% endif %}
tests/functional/clock-trigger-inexact/00-big-offset/flow.cylc:# Test clock-trigger offset involving inexact intervals (months and years),
tests/functional/clock-trigger-inexact/00-big-offset/flow.cylc:        clock-trigger = baz({{OFFSET}})  # old-style clock-triggered task
tests/functional/cylc-config/00-simple/section1.stdout:    clock-trigger =
tests/functional/cylc-show/clock-triggered-non-utc-mode/flow.cylc:        clock-trigger = foo(PT5M)
tests/functional/runahead/release-update/flow.cylc:        clock-trigger = foo(P0Y)
tests/functional/special/07-clock-triggered-360.t:grep_ok "WorkflowConfigError: clock-trigger tasks require \[scheduling\]cycling mode=" \
tests/functional/special/clock-360/flow.cylc:        clock-trigger = clock(0)
tests/functional/validate/35-pass-special-tasks-non-word-names.t:        clock-trigger = t-1, t+1, t%1, t@1
tests/functional/validate/37-special-implicit-task.t:        clock-trigger = foo(PT0M)
```

### CODE REVIEW

**Comment by hjoliver** [comment]
Two functional tests deleted because they only tested `cylc show` output for old-style clock-triggers.

**Comment by oliver-sanders** [comment]
That was quick!

**Review by oliver-sanders — APPROVED** [review]
Nice, tested as working.

**Review by MetRonnie — APPROVED** [review]
LGTM (just codecov falling over as is its wont)

File: cylc/flow/scripts/show.py [hunk 1]
```diff
@@ -327,25 +323,18 @@ async def prereqs_and_outputs_query(
...
-                            'Clock trigger time reached',
-                            state)
-                        print(f'  o Triggers at ... {time_str}')
                     for ext_trig in t_proxy['externalTriggers']:
                         state = ext_trig['satisfied']
                         print_msg_state(
                             f'{ext_trig["label"]} ... {state}',
                             state)
                     for xtrig in t_proxy['xtriggers']:
+                        print(xtrig)
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/scripts/show.py:337]: @hjoliver one question here, otherwise good to go.

**Pull request merged**